### PR TITLE
Introduce decorator: wrap and autowrap

### PR DESCRIPTION
**Pull request by wookayin**
A prototype implementation of #3.

* `tfplot.wrap`: It can be used as a decorator as well as with a normal function call.
* `tfplot.autowrap`: Similar to `wrap`, but supports additional features (e.g. automatic injection of `fig`, `ax`, etc.)

This PR is for reviewing.

### CODE REVIEW

File: tfplot/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+from matplotlib.figure import Figure
+from matplotlib.axes import Axes
+
+
+# a dummy marker object to avoid pylint issues on decorators.
+REQUIRED = object()
+
+
+@biwrap
+def wrap(plot_func=REQUIRED, _sentinel=None,
```
Comment (ferrine) [inline tfplot/wrapper.py]: Why not using keyword only arguments? Compatibility?

File: tfplot/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+        fig, ax = figure.subplots(figsize=_figsize)
+        return fig, ax
+
+    # Decorates `plot_func` with additional aspects
+    # (e.g. auto-injection, return value handling)
+    @functools.wraps(plot_func)
+    def _wrapped_plot_fn(*args, **kwargs_call):
+        # (1) auto-inject fig, ax
+        if fig_ax_mode:
```
Comment (ferrine) [inline tfplot/wrapper.py]: This is a bit hard to follow

File: tfplot/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+@biwrap
+def autowrap(plot_func=REQUIRED, _sentinel=None,
+             batch=False, name=None, figsize=None, tight_layout=False):
+    """
+    Wrap a function as a TensorFlow operation similar to :func:`tfplot.wrap()`
+    (as a decorator or with normal function call), but provides with additional
+    features such as auto-creating matplotlib figures.
+
+      - (``fig``, ``ax``) matplotlib objects are automatically created and
```
Comment (ferrine) [inline tfplot/wrapper.py]: Nice, well documented

**Review by ferrine — COMMENTED** [review]
Well, this code is fine, and well documented. I do not have big concerns on design as it seems to be quite agnostic to the function used

File: tfplot/wrapper.py (hunk 1 above)
Comment (wookayin) [reply tfplot/wrapper.py]: I want to support python 2.7 for the time being (See TF's cross entropy loss function).

**Comment by wookayin** [comment]
@ferrine Thanks for your comments!
